lib/info-extras.js

```diff
@@ -124,11 +124,46 @@ exports.getPublished = (body) => {
  * @return {Array.<Object>}
  */
 exports.getRelatedVideos = (body) => {
-  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': {"rvs":', '},');
+  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': ', ',\n');
+  let watchNextJson;
   try {
     jsonStr = JSON.parse(jsonStr);
-  } catch (err) {
+    watchNextJson = JSON.parse(jsonStr.watch_next_response);
+  }
+  catch (err) {
     return [];
   }
-  return jsonStr.split(',').map((link) => qs.parse(link));
-};
+  let rvsParams = jsonStr.rvs.split(',').map((e) => qs.parse(e));
+  let secondaryResults = watchNextJson.contents.twoColumnWatchNextResults.secondaryResults.secondaryResults.results;
+  let recomendedVideos = [];
+  for (let result of secondaryResults) {
+    let details = result.compactVideoRenderer;
+    if (details) {
+      try {
+        let viewCount = details.viewCountText.simpleText;
+        if (!viewCount) {
+          viewCount = details.viewCountText.runs[0].text.replace(' watching', '');
+        }
+        let shortViewCount = details.shortViewCountText.simpleText;
+        if (!shortViewCount.match(/^\d/.test()) || !shortViewCount)
+        {
+          if (shortViewCount = rvsParams.find((elem) => elem.id === details.videoId)) {
+            shortViewCount = shortViewCount.short_view_count_text;
+          }
+        }
+        recomendedVideos.push({
+          id: details.videoId,
+          title: details.title.simpleText,
+          author: details.shortBylineText.runs[0].text,
+          ucid: details.shortBylineText.runs[0].navigationEndpoint.browseEndpoint.browseId,
+          author_thumbnail: details.channelThumbnail.thumbnails[0].url,
+          short_view_count_text: shortViewCount.split(' ')[0],
+          view_count: viewCount.match(/^\d/.test()) ? viewCount.split(' ')[0] : shortViewCount.split(' ')[0],
```

Comment:
use `.test()` here too.

don't repeat `.split(' ')[0]`, put the object that the conditional operator switches out in parenthesis and the `.split(...` outside it to prevent repeated code